src/test/java/uk/gov/pay/directdebit/payments/services/sandbox/SandboxPaymentStateCalculatorTest.java

```diff
@@ -16,54 +23,123 @@
 import uk.gov.pay.directdebit.payments.model.SandboxPaymentId;
 import uk.gov.pay.directdebit.webhook.sandbox.resources.WebhookSandboxResource.SandboxEventAction;
 
+import java.time.ZonedDateTime;
 import java.util.Optional;
 
+import static java.time.ZoneOffset.UTC;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 import static org.mockito.BDDMockito.given;
+import static uk.gov.pay.directdebit.events.model.GovUkPayEventType.PAYMENT_SUBMITTED;
 import static uk.gov.pay.directdebit.gatewayaccounts.model.PaymentProvider.SANDBOX;
 import static uk.gov.pay.directdebit.mandate.fixtures.MandateFixture.aMandateFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture.aGatewayAccountFixture;
+import static uk.gov.pay.directdebit.payments.fixtures.GovUkPayEventFixture.aGovUkPayEventFixture;
 import static uk.gov.pay.directdebit.payments.fixtures.PaymentFixture.aPaymentFixture;
-import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxPaymentStateCalculator.SANDBOX_ACTIONS_THAT_CHANGE_STATE;
+import static uk.gov.pay.directdebit.payments.services.GovUkPayEventToPaymentStateMapper.GOV_UK_PAY_EVENT_TYPES_THAT_CHANGE_PAYMENT_STATE;
+import static uk.gov.pay.directdebit.payments.services.sandbox.SandboxEventToPaymentStateMapper.SANDBOX_ACTIONS_THAT_CHANGE_PAYMENT_STATE;
 
-@RunWith(MockitoJUnitRunner.class)
+@RunWith(JUnitParamsRunner.class)
 public class SandboxPaymentStateCalculatorTest {
 
     private static final SandboxPaymentId SANDBOX_PAYMENT_ID = SandboxPaymentId.valueOf("Sandbox payment ID");
 
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
     @Mock
     private SandboxEvent mockSandboxEvent;
 
     @Mock
     private SandboxEventDao mockSandboxEventDao;
 
+    @Mock
+    private GovUkPayEventDao mockGovUkPayEventDao;
+
     @InjectMocks
     private SandboxPaymentStateCalculator sandboxPaymentStateCalculator;
-    
+
     private Payment payment;
-    
+
     @Before
     public void setUp() {
         GatewayAccountFixture gatewayAccountFixture = aGatewayAccountFixture()
                 .withPaymentProvider(SANDBOX);
 
         MandateFixture mandateFixture = aMandateFixture()
                 .withGatewayAccountFixture(gatewayAccountFixture);
-        
+
         payment = aPaymentFixture()
                 .withMandateFixture(mandateFixture)
                 .withPaymentProviderId(SANDBOX_PAYMENT_ID)
                 .toEntity();
+    }
+
+    @Test
+    @Parameters({
+            "PAID_OUT, SUCCESS"
+    })
+    public void sandboxEventActionMapsToState(String action, String expectedState) {
+        given(mockSandboxEvent.getEventAction()).willReturn(action);
```

Comment:
Out of interest why did you opt to mock the sandbox event rather than create one with the builder. Not saying I think its right or wrong just curious although there is now a mix of creating events and using mocks (see `govUkPayEventTypeMapsToState` which creates a `GovUkPayEvent`.
```
var sandboxEvent = aSandboxEvent()
                .withEventAction(action)
                .build();
```

Reply:
Oh yeh I'll do that. I kept the mock behaviour because there was no fixture like there is for other events and wasn't sure it was worthwhile adding one. But as this doesn't need to be inserted into the database I can just use the builder on the SandboxEvent. Good suggestion 👍

Reply:
changed